src/outputs/output_utils.hpp

```diff
@@ -215,32 +293,35 @@ std::vector<T> FlattenBlockInfo(Mesh *pm, int shape, Function_t f) {
 }
 
 // mirror must be provided because copying done externally
-template <typename Data_t, typename idx_t, typename Function_t>
-void PackOrUnpackVar(MeshBlock *pmb, Variable<Real> *pvar, bool do_ghosts, idx_t &idx,
-                     std::vector<Data_t> &data, Function_t f) {
-  const auto &Nt = pvar->GetDim(6);
-  const auto &Nu = pvar->GetDim(5);
-  const auto &Nv = pvar->GetDim(4);
+template <typename idx_t, typename Function_t>
+void PackOrUnpackVar(const VarInfo &info, bool do_ghosts, idx_t &idx, Function_t f) {
   const IndexDomain domain = (do_ghosts ? IndexDomain::entire : IndexDomain::interior);
-  IndexRange kb, jb, ib;
-  if (pvar->metadata().Where() == MetadataFlag(Metadata::Cell)) {
-    kb = pmb->cellbounds.GetBoundsK(domain);
-    jb = pmb->cellbounds.GetBoundsJ(domain);
-    ib = pmb->cellbounds.GetBoundsI(domain);
-    // TODO(JMM): Add topological elements here
-  } else { // metadata none
-    kb = {0, pvar->GetDim(3) - 1};
-    jb = {0, pvar->GetDim(2) - 1};
-    ib = {0, pvar->GetDim(1) - 1};
+  // shape as written to or read from. contains additional padding
+  // in orthogonal directions.
+  // e.g., Face1-centered var is shape (N1+1)x(N2+1)x(N3+1)
+  // format is
+  // topological_elems x tensor_elems x block_elems
```

Comment:
Followup up on my comment above: I see, so it mirrors our pattern of allocation too much in order to reduce complexity with different shapes.

Reply:
that's right---so we output more data than we need to, but it simplifies things.